src/base/bittorrent/session.cpp

```diff
@@ -2113,7 +2113,7 @@ bool Session::addTorrent(const TorrentInfo &torrentInfo, const AddTorrentParams
 bool Session::addTorrent_impl(CreateTorrentParams params, const MagnetUri &magnetUri,
                               TorrentInfo torrentInfo, const QByteArray &fastresumeData)
 {
-#if (LIBTORRENT_VERSION_NUM < 10200)
+#if (LIBTORRENT_VERSION_NUM < 10200) || (LIBTORRENT_VERSION_NUM >= 20000)
```

Comment:
This part of code is targeting older *and* newest libtorrent versions, is it really correct?

Reply:
I don't think it's correct. It is unlikely that libtorrent-2.0 codepath can coincide with libtorrent-1.1 one.

Reply:
Generally I think we shouldn't merge it until this "feature" is implemented in a libtorrent-2.0 way. I intend to do it later, unless @arvidn wants to do it himself.

Reply:
I'm pretty sure this is correct. It's targeting a libtorrent that doesn't support the custom storage. It's not clear to me why the custom storage was limited to libtorrent 1.2 and later, but I'm sure there's a reason.

Reply:
>It's targeting a libtorrent that doesn't support the custom storage.

This (very long if you didn't notice it) part of code targeting libtorrent-1.1 related logic of adding torrent to the session.
>It's not clear to me why the custom storage was limited to libtorrent 1.2 and later,

Since I don't want to implement it for outdated libtorrent version (I have no time to test and maintain it there) just before we drop to support it completely.

Reply:
@glassez 
I tested compiling qbt with libt 2.0 with deprecated functions ON and the custom storage issue is actually the major part that is blocking our way. It would be best if you can take a look and then our master branch can have basic support of libt 2.0.
ps. I plan to migrate away from libt deprecated functions gradually and eventually.

Reply:
OK, I'll get to it soon.
In any case, you can just "disable" it for now to make sure everything else is at least compilable.